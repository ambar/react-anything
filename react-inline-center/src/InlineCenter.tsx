// zero-width space
const ZWSP = '\u200b'

type Props = {
  children: React.ReactNode
  disabled?: boolean
}

const InlineCenter = ({children, disabled = false}: Props) =>
  disabled ? (
    <>{children}</>
  ) : (
    <span style={{display: 'inline-flex', alignItems: 'center'}}>
      {ZWSP}
      {children}
    </span>
  )

export default InlineCenter
