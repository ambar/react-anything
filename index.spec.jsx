import React from 'react'
import renderer from 'react-test-renderer'
import InlineCenter from './index'

test('renders default', () => {
  const tree = renderer.create(<InlineCenter>text</InlineCenter>).toJSON()
  expect(tree).toMatchSnapshot()
})

test('renders disabled', () => {
  const tree = renderer.create(<InlineCenter disabled>text</InlineCenter>).toJSON()
  expect(tree).toMatchSnapshot()
})
