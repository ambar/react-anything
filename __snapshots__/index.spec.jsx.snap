// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default 1`] = `
<span
  style={
    Object {
      "alignItems": "center",
      "display": "inline-flex",
    }
  }
>
  â€‹
  text
</span>
`;

exports[`renders disabled 1`] = `"text"`;
